web3/_utils/method_formatters.py

```diff
@@ -754,42 +754,87 @@ def raise_transaction_not_found_with_index(
 }
 
 
-def filter_wrapper(
-    module: "Eth",
-    method: RPCEndpoint,
-    filter_id: HexStr,
-) -> Union[BlockFilter, TransactionFilter, LogFilter]:
-    if method == RPC.eth_newBlockFilter:
-        return BlockFilter(filter_id, eth_module=module)
-    elif method == RPC.eth_newPendingTransactionFilter:
-        return TransactionFilter(filter_id, eth_module=module)
-    elif method == RPC.eth_newFilter:
-        return LogFilter(filter_id, eth_module=module)
-    else:
-        raise NotImplementedError(
-            "Filter wrapper needs to be used with either "
-            f"{RPC.eth_newBlockFilter}, {RPC.eth_newPendingTransactionFilter}"
-            f" or {RPC.eth_newFilter}"
-        )
+# def filter_wrapper(
+#     module: "Eth",
+#     method: RPCEndpoint,
+#     filter_id: HexStr,
+# ) -> Union[BlockFilter, TransactionFilter, LogFilter]:
+#     if method == RPC.eth_newBlockFilter:
+#         return BlockFilter(filter_id, eth_module=module)
+#     elif method == RPC.eth_newPendingTransactionFilter:
+#         return TransactionFilter(filter_id, eth_module=module)
+#     elif method == RPC.eth_newFilter:
+#         return LogFilter(filter_id, eth_module=module)
+#     else:
+#         raise NotImplementedError(
+#             "Filter wrapper needs to be used with either "
+#             f"{RPC.eth_newBlockFilter}, {RPC.eth_newPendingTransactionFilter}"
+#             f" or {RPC.eth_newFilter}"
+#         )
 
 
-def async_filter_wrapper(
-    module: "AsyncEth",
+def filter_wrapper(
+    module: Union["AsyncEth", "Eth"],
     method: RPCEndpoint,
     filter_id: HexStr,
-) -> Union[AsyncBlockFilter, AsyncTransactionFilter, AsyncLogFilter]:
-    if method == RPC.eth_newBlockFilter:
-        return AsyncBlockFilter(filter_id, eth_module=module)
-    elif method == RPC.eth_newPendingTransactionFilter:
-        return AsyncTransactionFilter(filter_id, eth_module=module)
-    elif method == RPC.eth_newFilter:
-        return AsyncLogFilter(filter_id, eth_module=module)
+) -> Union[
+    AsyncBlockFilter,
+    AsyncTransactionFilter,
+    AsyncLogFilter,
+    BlockFilter,
+    TransactionFilter,
+    LogFilter,
+]:
+    
+    if type(module).__name__ == "Eth":
```

Comment:
Sorry if this is just a wip and you don't want feedback, but could you check for `module.is_async` here?

Reply:
No problem - feedback always welcome, especially if I've asked for review. Will try it.

Reply:
Ok, updated, ready for re-review.